crates/serialize_hierarchy_derive/src/process_struct.rs

```diff
@@ -95,7 +94,7 @@ fn generate_serde_serialization(fields: &Fields) -> Vec<TokenStream> {
             let pattern = name.to_string();
             let error_message = format!("failed to serialize field `{name}`");
             Some(quote! {
-                #pattern => serde_json::to_value(&self.#name).wrap_err(#error_message)
+                #pattern => serialize_hierarchy::serde_json::to_value(&self.#name).wrap_err(#error_message)
```

Comment:
Schmiddy implemented that `serialize_hierarchy` now reexports `serde_json`. Since we have a monorepo and workspace dependencies, it does not make that much sense, but I kept it because it is good practice for reusable crates to reexport their types used in their public interfaces.